libdnf/solv/solver.hpp

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_SOLV_SOLVER_HPP
+#define LIBDNF_SOLV_SOLVER_HPP
+
+#include "id_queue.hpp"
+#include "pool.hpp"
+
+#include <filesystem>
+
+extern "C" {
+#include <solv/solver.h>
+}
+
+
+namespace libdnf::solv {
+
+class Solver {
+public:
+    Solver(){};
+    explicit Solver(Pool & pool);
+
+    Solver(const Solver & solver) = delete;
+    Solver & operator=(const Solver & solver) = delete;
+
+    ~Solver();
+
+    ::Solver * operator*() { return &*solver; }
+    ::Solver * operator*() const { return &*solver; }
+
+    ::Solver * operator->() { return &*solver; }
+    ::Solver * operator->() const { return &*solver; }
+
+    /// initialize or reinitialize `solver` with given pool
+    void init(Pool & pool);
+
+    /// Used in ModuleGoalPrivate::resolve()
+    /// If ModuleSack::Impl moves from libsolv `Pool *` to `libdnf::solv::Pool`,
+    /// this method can be dropped.
+    void init(::Pool * pool);
+
+    /// drop current `solver` object
+    void reset();
+
+    /// returns true if `solver` exists
+    bool is_initialized() { return solver != nullptr; };
+
+    /// Write solver debug data to given directory
+    /// @param with_transaction Whether transaction data are dumped
```

Comment:
I am sorry, I do not understand the description or a use case.

Reply:
If you mean the with_transaction parameter, it's used in `libdnf/repo/repo_sack.cpp` in this PR.
If true, it sets `TESTCASE_RESULT_TRANSACTION | TESTCASE_RESULT_PROBLEMS` flags in libsolv's testcase_write call.